### DISCUSSION TITLE
eln validator

### DISCUSSION THREAD

**Discussion by SteffenBrinckmann** [TheELNConsortium/TheELNFileFormat#12] [Ideas]
Hey everybody,
I was thinking, by having a tool that transforms eln into an Readme-Markdown output, we have done the first step towards a validator, if it is valid the transformation works.
Why not have a validator for this version 1: checks if basic format is followed.
If you agree, I could use the transformer as starting point and then pepper it with if-statements. Let me know if you would find that helpful.

**Comment by NicolasCARPi** [comment]
Maybe adding a `check` command to this project would make sense? https://github.com/ResearchObject/ro-crate-py

**Reply by NicolasCARPi** [reply]
There are two things: checking the validity of the ro-crate for which we could use pre-existing software, and checking the validity of the .eln, for which we have to write our own.

**Comment by SteffenBrinckmann** [comment]
Great ideas!
- I went on that github repo and supported a feature request to add a validator
- For the meantime, I would see if that software can open&list the content (that is also some validation); and write some other validation, as mentioned in your last comment.

**Comment by SteffenBrinckmann** [comment]
First version is there, according to that process: ro-crate.py and then custom tests.
Please check if there are holes in the validator.
